sentry/src/main/java/io/sentry/util/CheckInUtils.java

```diff
@@ -30,12 +31,11 @@ public static <U> U withCheckIn(
       final @Nullable MonitorConfig monitorConfig,
       final @NotNull Callable<U> callable)
       throws Exception {
+    final @NotNull ISentryLifecycleToken lifecycleToken = Sentry.pushIsolationScope();
```

Comment:
Maybe this should also be configurable to use `pushScope` or `pushIsolationScope`.

Reply:
Default should probably also be isolated. Add overload with `isolated: true/false`, defaulting to true